[1.20.1] 1.1.2 Crashes on Quilt

https://mclo.gs/zsF1lLY

```
---- Minecraft Crash Report ----
// Oh - I know what I did wrong!
Time: 2023-07-17 13:29:12
Description: Initializing game
java.lang.NoSuchMethodError: 'org.quiltmc.qsl.resource.loader.api.ResourcePackActivationType net.minecraft.class_3262.getActivationType()'
	at net.minecraft.class_3288.handler$zco000$quilt_resource_loader$quilt$onInit(class_3288.java:545)
	at net.minecraft.class_3288.<init>(class_3288.java:66)
	at net.minecraft.class_3288.method_14456(class_3288.java:52)
	at net.minecraft.class_3288.method_45275(class_3288.java:48)
	at net.minecraft.class_1065.method_45257(class_1065.java:88)
	at net.minecraft.class_7678.method_14453(class_7678.java:39)
	at net.minecraft.class_3283.method_29212(class_3283.java:39)
	at net.minecraft.class_3283.method_14445(class_3283.java:32)
	at net.minecraft.class_310.<init>(class_310.java:513)
	at net.minecraft.client.main.Main.main(Main.java:211)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:524)
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:82)
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------
-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_3288.handler$zco000$quilt_resource_loader$quilt$onInit(class_3288.java:545)
	at net.minecraft.class_3288.<init>(class_3288.java:66)
	at net.minecraft.class_3288.method_14456(class_3288.java:52)
	at net.minecraft.class_3288.method_45275(class_3288.java:48)
	at net.minecraft.class_1065.method_45257(class_1065.java:88)
	at net.minecraft.class_7678.method_14453(class_7678.java:39)
	at net.minecraft.class_3283.method_29212(class_3283.java:39)
	at net.minecraft.class_3283.method_14445(class_3283.java:32)
	at net.minecraft.class_310.<init>(class_310.java:513)
-- Initialization --
Details:
	Modules: 
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:211)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:524)
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:82)
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
```

+1 same, here's my stack trace with MixinTrace:
```
Mod Table Version: 2
Plugin Types: {quilted_fabric_loader=[Fabric], quilt_loader=[Quilt, Builtin]}
[16:35:38] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/C:/Users/lilit/AppData/Roaming/PrismLauncher/libraries/net/fabricmc/sponge-mixin/0.12.5+mixin.0.8.5/sponge-mixin-0.12.5+mixin.0.8.5.jar Service=Knot/Quilt Env=CLIENT
[16:35:38] [main/INFO]: Compatibility level set to JAVA_17
[16:35:38] [main/ERROR]: Mixin prepare for mod quilt_resource_loader failed preparing ResourcePackMixin in #quilt_resource_loader:quilt_resource_loader.mixins.json: org.spongepowered.asm.mixin.transformer.throwables.MixinTargetAlreadyLoadedException Critical problem: #quilt_resource_loader:quilt_resource_loader.mixins.json:ResourcePackMixin from mod quilt_resource_loader target net.minecraft.class_3262 was loaded too early.
org.spongepowered.asm.mixin.transformer.throwables.MixinTargetAlreadyLoadedException: Critical problem: #quilt_resource_loader:quilt_resource_loader.mixins.json:ResourcePackMixin from mod quilt_resource_loader target net.minecraft.class_3262 was loaded too early.
	at org.spongepowered.asm.mixin.transformer.MixinInfo.readDeclaredTargets(MixinInfo.java:948) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinInfo.<init>(MixinInfo.java:882) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepareMixins(MixinConfig.java:852) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepare(MixinConfig.java:777) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:540) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.select(MixinProcessor.java:462) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.checkSelect(MixinProcessor.java:438) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:290) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:453) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:244) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadIntoTarget(KnotClassLoader.java:243) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.init(Knot.java:167) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:76) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28) ~[quilt-loader-0.19.2.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71) ~[NewLaunch.jar:?]
[16:35:38] [main/WARN]: Mixin transformation of org.quiltmc.loader.impl.launch.knot.UnusedEmptyTargetClass failed
org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [#quilt_resource_loader:quilt_resource_loader.mixins.json:ResourcePackMixin from mod quilt_resource_loader] from phase [DEFAULT] in config [#quilt_resource_loader:quilt_resource_loader.mixins.json] FAILED during PREPARE
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:638) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinPrepareError(MixinProcessor.java:585) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:543) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.select(MixinProcessor.java:462) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.checkSelect(MixinProcessor.java:438) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:290) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:453) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:244) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadIntoTarget(KnotClassLoader.java:243) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.init(Knot.java:167) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:76) ~[quilt-loader-0.19.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28) ~[quilt-loader-0.19.2.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71) ~[NewLaunch.jar:?]
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTargetAlreadyLoadedException: Critical problem: #quilt_resource_loader:quilt_resource_loader.mixins.json:ResourcePackMixin from mod quilt_resource_loader target net.minecraft.class_3262 was loaded too early.
	at org.spongepowered.asm.mixin.transformer.MixinInfo.readDeclaredTargets(MixinInfo.java:948) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinInfo.<init>(MixinInfo.java:882) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepareMixins(MixinConfig.java:852) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepare(MixinConfig.java:777) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:540) ~[sponge-mixin-0.12.5+mixin.0.8.5.jar:0.12.5+mixin.0.8.5]
	... 14 more
Exception caught from launcher
java.lang.RuntimeException: Mixin transformation of org.quiltmc.loader.impl.launch.knot.UnusedEmptyTargetClass failed
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:458)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:244)
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadIntoTarget(KnotClassLoader.java:243)
	at org.quiltmc.loader.impl.launch.knot.Knot.init(Knot.java:167)
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:76)
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [#quilt_resource_loader:quilt_resource_loader.mixins.json:ResourcePackMixin from mod quilt_resource_loader] from phase [DEFAULT] in config [#quilt_resource_loader:quilt_resource_loader.mixins.json] FAILED during PREPARE
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:638)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinPrepareError(MixinProcessor.java:585)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:543)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.select(MixinProcessor.java:462)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.checkSelect(MixinProcessor.java:438)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:290)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:453)
	... 8 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTargetAlreadyLoadedException: Critical problem: #quilt_resource_loader:quilt_resource_loader.mixins.json:ResourcePackMixin from mod quilt_resource_loader target net.minecraft.class_3262 was loaded too early.
	at org.spongepowered.asm.mixin.transformer.MixinInfo.readDeclaredTargets(MixinInfo.java:948)
	at org.spongepowered.asm.mixin.transformer.MixinInfo.<init>(MixinInfo.java:882)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepareMixins(MixinConfig.java:852)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepare(MixinConfig.java:777)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:540)
	... 14 more
Exiting with ERROR
Process exited with code 2.
```

--- a/build.gradle
+++ b/build.gradle
@@ -1,6 +1,6 @@
 plugins {
     id "architectury-plugin" version "3.4-SNAPSHOT"
-    id "dev.architectury.loom" version "1.2-SNAPSHOT" apply false
+    id "dev.architectury.loom" version "1.3.+" apply false
     id 'io.github.juuxel.loom-vineflower' version '1.11.0' apply false
 }
 

--- a/common/build.gradle
+++ b/common/build.gradle
@@ -10,7 +10,7 @@ dependencies {
     // We depend on fabric loader here to use the fabric @Environment annotations and get the mixin dependencies
     // Do NOT use other classes from fabric loader
     modImplementation "net.fabricmc:fabric-loader:${rootProject.fabric_loader_version}"
-    include(implementation("blue.endless:jankson:${rootProject.jankson_version}"))
+    implementation("blue.endless:jankson:${rootProject.jankson_version}")
 
     modImplementation "com.github.glitchfiend:TerraBlender-fabric:${rootProject.minecraft_version}-${rootProject.terrablender_version}"
 }

--- a/common/src/main/java/net/cristellib/CristelLibExpectPlatform.java
+++ b/common/src/main/java/net/cristellib/CristelLibExpectPlatform.java
@@ -35,16 +35,6 @@ public class CristelLibExpectPlatform {
     }
 
     @ExpectPlatform
-    public static boolean isModLoadedWithVersion(String modid, String minVersion){
-        throw new AssertionError();
-    }
-
-    @ExpectPlatform
-    public static boolean isModLoaded(String modid){
-        throw new AssertionError();
-    }
-
-    @ExpectPlatform
     public static Map<String, Set<StructureConfig>> getConfigs(CristelLibRegistry registry) {
         throw new AssertionError();
     }

--- a/common/src/main/java/net/cristellib/ModLoadingUtil.java
+++ b/common/src/main/java/net/cristellib/ModLoadingUtil.java
@@ -1,0 +1,15 @@
+package net.cristellib;
+
+import dev.architectury.injectables.annotations.ExpectPlatform;
+
+public class ModLoadingUtil {
+    @ExpectPlatform
+    public static boolean isModLoadedWithVersion(String modid, String minVersion){
+        throw new AssertionError();
+    }
+
+    @ExpectPlatform
+    public static boolean isModLoaded(String modid){
+        throw new AssertionError();
+    }
+}

--- a/common/src/main/java/net/cristellib/data/Conditions.java
+++ b/common/src/main/java/net/cristellib/data/Conditions.java
@@ -3,7 +3,7 @@ package net.cristellib.data;
 import com.google.gson.JsonArray;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
-import net.cristellib.CristelLibExpectPlatform;
+import net.cristellib.ModLoadingUtil;
 
 public class Conditions {
 
@@ -23,7 +23,7 @@ public class Conditions {
     public static boolean readCondition(JsonObject object){
         String type = object.get("type").getAsString();
         if(type.equals("mod_loaded")){
-            return CristelLibExpectPlatform.isModLoaded(object.get("mod").getAsString());
+            return ModLoadingUtil.isModLoaded(object.get("mod").getAsString());
         }
 
         return false;

--- a/common/src/main/java/net/cristellib/mixin/MixinTerrablenderConfig.java
+++ b/common/src/main/java/net/cristellib/mixin/MixinTerrablenderConfig.java
@@ -7,7 +7,7 @@ import org.spongepowered.asm.mixin.injection.Redirect;
 import terrablender.config.Config;
 import terrablender.config.TerraBlenderConfig;
 
-@Mixin(TerraBlenderConfig.class)
+@Mixin(value = TerraBlenderConfig.class, remap = false)
 public class MixinTerrablenderConfig {
     @Redirect(method = "<init>", at = @At(value = "INVOKE", target = "Lterrablender/config/Config;addNumber(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Number;Ljava/lang/Number;Ljava/lang/Number;)Ljava/lang/Number;"))
     private <T extends Number & Comparable<T>> Number create(Config instance, String comment, String key, T defaultValue, T min, T max) {

--- a/common/src/main/java/net/cristellib/mixin/TerrablenderPlugin.java
+++ b/common/src/main/java/net/cristellib/mixin/TerrablenderPlugin.java
@@ -1,6 +1,6 @@
 package net.cristellib.mixin;
 
-import net.cristellib.util.Util;
+import net.cristellib.util.TerrablenderUtil;
 import org.objectweb.asm.tree.ClassNode;
 import org.spongepowered.asm.mixin.extensibility.IMixinConfigPlugin;
 import org.spongepowered.asm.mixin.extensibility.IMixinInfo;
@@ -21,7 +21,7 @@ public class TerrablenderPlugin implements IMixinConfigPlugin {
 
     @Override
     public boolean shouldApplyMixin(String targetClassName, String mixinClassName) {
-        return Util.isTerrablenderLoaded();
+        return TerrablenderUtil.isModLoaded();
     }
 
     @Override

--- a/common/src/main/java/net/cristellib/util/TerrablenderUtil.java
+++ b/common/src/main/java/net/cristellib/util/TerrablenderUtil.java
@@ -1,5 +1,8 @@
 package net.cristellib.util;
 
+import net.cristellib.CristelLib;
+import net.cristellib.ModLoadingUtil;
+
 public class TerrablenderUtil {
 
     private static boolean enableMixin = false;
@@ -12,4 +15,7 @@ public class TerrablenderUtil {
         enableMixin = bl;
     }
 
+    public static boolean isModLoaded(){
+        return ModLoadingUtil.isModLoadedWithVersion("terrablender", CristelLib.minTerraBlenderVersion);
+    }
 }

--- a/common/src/main/java/net/cristellib/util/Util.java
+++ b/common/src/main/java/net/cristellib/util/Util.java
@@ -1,7 +1,5 @@
 package net.cristellib.util;
 
-import net.cristellib.CristelLib;
-import net.cristellib.CristelLibExpectPlatform;
 import net.minecraft.resources.ResourceLocation;
 
 import java.util.Map;
@@ -20,10 +18,6 @@ public class Util {
                 addTo.put(string, addFrom.get(string));
             }
         }
-    }
-
-    public static boolean isTerrablenderLoaded(){
-        return CristelLibExpectPlatform.isModLoadedWithVersion("terrablender", CristelLib.minTerraBlenderVersion);
     }
 
     public static ResourceLocation getLocationForStructureSet(ResourceLocation location){

--- a/fabric-like/src/main/java/net/cristellib/fabriclike/CristelLibFabricLikePlatform.java
+++ b/fabric-like/src/main/java/net/cristellib/fabriclike/CristelLibFabricLikePlatform.java
@@ -1,9 +1,6 @@
 package net.cristellib.fabriclike;
 
-import net.cristellib.CristelLib;
-import net.cristellib.CristelLibExpectPlatform;
-import net.cristellib.CristelLibRegistry;
-import net.cristellib.StructureConfig;
+import net.cristellib.*;
 import net.cristellib.api.CristelLibAPI;
 import net.cristellib.data.ReadData;
 import net.cristellib.util.Util;
@@ -11,8 +8,6 @@ import net.fabricmc.fabric.api.resource.ResourcePackActivationType;
 import net.fabricmc.fabric.impl.resource.loader.ModNioResourcePack;
 import net.fabricmc.loader.api.FabricLoader;
 import net.fabricmc.loader.api.ModContainer;
-import net.fabricmc.loader.api.Version;
-import net.fabricmc.loader.api.VersionParsingException;
 import net.minecraft.network.chat.Component;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.packs.PackResources;
@@ -37,21 +32,6 @@ public class CristelLibFabricLikePlatform {
             CristelLib.LOGGER.warn("Couldn't get mod container for modid: " + modid);
             return null;
         }
-    }
-
-    public static boolean isModLoadedWithVersion(String modid, String minVersion) {
-        if(CristelLibExpectPlatform.isModLoaded(modid)){
-            Version version = FabricLoader.getInstance().getModContainer(modid).get().getMetadata().getVersion();
-            Version min;
-            try {
-                min = Version.parse(minVersion);
-            } catch (VersionParsingException e) {
-                CristelLib.LOGGER.error("Couldn't parse version: " + minVersion);
-                return false;
-            }
-            return version.compareTo(min) >= 0;
-        }
-        return false;
     }
 
     public static @Nullable Path getResourceDirectory(String modid, String subPath) {

--- a/fabric-like/src/main/java/net/cristellib/fabriclike/ModLoadingUtilFabricLike.java
+++ b/fabric-like/src/main/java/net/cristellib/fabriclike/ModLoadingUtilFabricLike.java
@@ -1,0 +1,24 @@
+package net.cristellib.fabriclike;
+
+import net.cristellib.CristelLib;
+import net.cristellib.ModLoadingUtil;
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.Version;
+import net.fabricmc.loader.api.VersionParsingException;
+
+public class ModLoadingUtilFabricLike {
+    public static boolean isModLoadedWithVersion(String modid, String minVersion) {
+        if (ModLoadingUtil.isModLoaded(modid)) {
+            Version version = FabricLoader.getInstance().getModContainer(modid).get().getMetadata().getVersion();
+            Version min;
+            try {
+                min = Version.parse(minVersion);
+            } catch (VersionParsingException e) {
+                CristelLib.LOGGER.error("Couldn't parse version: " + minVersion);
+                return false;
+            }
+            return version.compareTo(min) >= 0;
+        }
+        return false;
+    }
+}

--- a/fabric/src/main/java/net/cristellib/fabric/CristelLibExpectPlatformImpl.java
+++ b/fabric/src/main/java/net/cristellib/fabric/CristelLibExpectPlatformImpl.java
@@ -4,7 +4,6 @@ import net.cristellib.CristelLibRegistry;
 import net.cristellib.StructureConfig;
 import net.cristellib.fabriclike.CristelLibFabricLikePlatform;
 import net.cristellib.util.Platform;
-import net.fabricmc.loader.api.FabricLoader;
 import net.minecraft.network.chat.Component;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.packs.PackResources;
@@ -35,14 +34,6 @@ public class CristelLibExpectPlatformImpl {
 
     public static List<Path> getRootPaths(String modId) {
         return CristelLibFabricLikePlatform.getRootPaths(modId);
-    }
-
-    public static boolean isModLoaded(String modId) {
-        return FabricLoader.getInstance().isModLoaded(modId);
-    }
-
-    public static boolean isModLoadedWithVersion(String modid, String minVersion) {
-        return CristelLibFabricLikePlatform.isModLoadedWithVersion(modid, minVersion);
     }
 
     public static Platform getPlatform() {

--- a/fabric/src/main/java/net/cristellib/fabric/ModLoadingUtilImpl.java
+++ b/fabric/src/main/java/net/cristellib/fabric/ModLoadingUtilImpl.java
@@ -1,0 +1,15 @@
+package net.cristellib.fabric;
+
+import net.cristellib.fabriclike.ModLoadingUtilFabricLike;
+import net.fabricmc.loader.api.FabricLoader;
+
+public class ModLoadingUtilImpl {
+
+    public static boolean isModLoaded(String modId) {
+        return FabricLoader.getInstance().isModLoaded(modId);
+    }
+
+    public static boolean isModLoadedWithVersion(String modid, String minVersion) {
+        return ModLoadingUtilFabricLike.isModLoadedWithVersion(modid, minVersion);
+    }
+}

--- a/forge/src/main/java/net/cristellib/forge/CristelLibExpectPlatformImpl.java
+++ b/forge/src/main/java/net/cristellib/forge/CristelLibExpectPlatformImpl.java
@@ -27,6 +27,8 @@ import javax.annotation.Nullable;
 import java.nio.file.Path;
 import java.util.*;
 
+import static net.cristellib.forge.ModLoadingUtilImpl.getPreLoadedModInfo;
+
 public class CristelLibExpectPlatformImpl {
 
     public static Path getConfigDirectory() {
@@ -132,47 +134,6 @@ public class CristelLibExpectPlatformImpl {
             file = container.getModInfo().getOwningFile().getFile();
         }
         return Collections.singletonList(file.getSecureJar().getRootPath());
-    }
-
-    public static boolean isModLoaded(String modid) {
-        ModList modList = ModList.get();
-        if(modList != null){
-            return modList.isLoaded(modid);
-        }
-        return isModPreLoaded(modid);
-    }
-
-    public static boolean isModPreLoaded(String modid) {
-        return getPreLoadedModInfo(modid) != null;
-    }
-
-    public static @Nullable ModInfo getPreLoadedModInfo(String modId){
-        for(ModInfo info : LoadingModList.get().getMods()){
-            if(info.getModId().equals(modId)) {
-                return info;
-            }
-        }
-        return null;
-    }
-
-    public static boolean isModLoadedWithVersion(String modid, String minVersion) {
-        if(isModLoaded(modid)){
-            ModList modList = ModList.get();
-            ArtifactVersion version;
-            if(modList != null) version = modList.getModContainerById(modid).get().getModInfo().getVersion();
-            else version = getPreLoadedModVersion(modid);
-
-            ArtifactVersion min;
-            min = new DefaultArtifactVersion(minVersion);
-            return version.compareTo(min) >= 0;
-        }
-        return false;
-    }
-
-    public static ArtifactVersion getPreLoadedModVersion(String modid) {
-        ModInfo info = getPreLoadedModInfo(modid);
-        if(info == null) throw new RuntimeException("Couldn't find mod: " + modid);
-        return info.getVersion();
     }
 
     public static Platform getPlatform() {

--- a/forge/src/main/java/net/cristellib/forge/ModLoadingUtilImpl.java
+++ b/forge/src/main/java/net/cristellib/forge/ModLoadingUtilImpl.java
@@ -1,0 +1,53 @@
+package net.cristellib.forge;
+
+import net.minecraftforge.fml.ModList;
+import net.minecraftforge.fml.loading.LoadingModList;
+import net.minecraftforge.fml.loading.moddiscovery.ModInfo;
+import org.apache.maven.artifact.versioning.ArtifactVersion;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+
+import javax.annotation.Nullable;
+
+public class ModLoadingUtilImpl {
+
+    public static boolean isModLoaded(String modid) {
+        ModList modList = ModList.get();
+        if(modList != null){
+            return modList.isLoaded(modid);
+        }
+        return isModPreLoaded(modid);
+    }
+
+    public static boolean isModPreLoaded(String modid) {
+        return getPreLoadedModInfo(modid) != null;
+    }
+
+    public static @Nullable ModInfo getPreLoadedModInfo(String modId){
+        for(ModInfo info : LoadingModList.get().getMods()){
+            if(info.getModId().equals(modId)) {
+                return info;
+            }
+        }
+        return null;
+    }
+
+    public static boolean isModLoadedWithVersion(String modid, String minVersion) {
+        if(isModLoaded(modid)){
+            ModList modList = ModList.get();
+            ArtifactVersion version;
+            if(modList != null) version = modList.getModContainerById(modid).get().getModInfo().getVersion();
+            else version = getPreLoadedModVersion(modid);
+
+            ArtifactVersion min;
+            min = new DefaultArtifactVersion(minVersion);
+            return version.compareTo(min) >= 0;
+        }
+        return false;
+    }
+
+    public static ArtifactVersion getPreLoadedModVersion(String modid) {
+        ModInfo info = getPreLoadedModInfo(modid);
+        if(info == null) throw new RuntimeException("Couldn't find mod: " + modid);
+        return info.getVersion();
+    }
+}

--- a/gradle.properties
+++ b/gradle.properties
@@ -12,8 +12,8 @@ fabric_api_version=0.89.2+1.20.2
 
 forge_version=1.20.2-48.0.13
 
-quilt_loader_version=0.20.2
-quilt_fabric_api_version=7.3.0+0.89.0-1.20.1
+quilt_loader_version=0.21.0
+quilt_fabric_api_version=7.3.1+0.89.3-1.20.1
 
-jankson_version = 1.2.1
+jankson_version=1.2.1
 terrablender_version=3.0.0.170

--- a/quilt/src/main/java/net/cristellib/fabric/CristelLibExpectPlatformImpl.java
+++ b/quilt/src/main/java/net/cristellib/fabric/CristelLibExpectPlatformImpl.java
@@ -97,14 +97,6 @@ public class CristelLibExpectPlatformImpl {
          */
     }
 
-    public static boolean isModLoaded(String modId) {
-        return QuiltLoader.isModLoaded(modId);
-    }
-
-    public static boolean isModLoadedWithVersion(String modid, String minVersion) {
-        return CristelLibFabricLikePlatform.isModLoadedWithVersion(modid, minVersion);
-    }
-
     public static Platform getPlatform() {
         return Platform.QUILT;
     }

--- a/quilt/src/main/java/net/cristellib/fabric/ModLoadingUtilImpl.java
+++ b/quilt/src/main/java/net/cristellib/fabric/ModLoadingUtilImpl.java
@@ -1,0 +1,15 @@
+package net.cristellib.fabric;
+
+import net.cristellib.fabriclike.ModLoadingUtilFabricLike;
+import org.quiltmc.loader.api.QuiltLoader;
+
+public class ModLoadingUtilImpl {
+
+    public static boolean isModLoaded(String modId) {
+        return QuiltLoader.isModLoaded(modId);
+    }
+
+    public static boolean isModLoadedWithVersion(String modid, String minVersion) {
+        return ModLoadingUtilFabricLike.isModLoadedWithVersion(modid, minVersion);
+    }
+}

